com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -0,0 +1,174 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudService;
+import org.cloudfoundry.client.lib.domain.CloudServiceInstance;
+import org.cloudfoundry.client.lib.domain.ServiceKey;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.client.XsCloudControllerClient;
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceOfferingExtended;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceInstanceGetter;
+import com.sap.cloud.lm.sl.cf.core.security.serialization.SecureSerializationFacade;
+import com.sap.cloud.lm.sl.cf.process.analytics.model.ServiceAction;
+import com.sap.cloud.lm.sl.common.util.CommonUtil;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+
+@Component("determineServiceCreateUpdateActionsStep")
+public class DetermineServiceCreateUpdateServiceActionsStep extends SyncFlowableStep {
+
+    @Inject
+    private ServiceInstanceGetter serviceInstanceGetter;
+
+    private SecureSerializationFacade secureSerializer = new SecureSerializationFacade();
+
+    @Override
+    protected StepPhase executeStep(ExecutionWrapper execution) throws Exception {
+        execution.getStepLogger().info("Entering determine state...");
```

Comment:
Extract "Entering determine state..." to message constant

Reply:
Debug message leftover.